File: flybanner/src/main/java/com/xia/flybanner/holder/FBHolder.java
package com.xia.flybanner.holder;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

/**
 * @author weixia
 * @date 2019/4/16.
 */
@SuppressWarnings("WeakerAccess")
public abstract class FBHolder<T> extends RecyclerView.ViewHolder {

    public FBHolder(View itemView) {
        super(itemView);
        initView(itemView);
    }

    protected abstract void initView(View itemView);

    public abstract void updateUI(T data);
}

File: flybanner/src/main/java/com/xia/flybanner/holder/FBHolder.kt
package com.xia.flybanner.holder

import android.view.View
import androidx.recyclerview.widget.RecyclerView

/**
 * @author weixia
 * @date 2019/4/16.
 */
abstract class FBHolder<T>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    init {
        apply {
            initView(itemView)
        }
    }

    protected abstract fun initView(itemView: View)

    abstract fun updateUI(data: T)
}
